server.js

```diff
@@ -1,33 +1,79 @@
-import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+//import data
+import profanityDictionary from './data/profanity-dictionary.json'
+import profanityCategories from './data/profanity-dictionary-categories.json'
+
+//import doc
+import documentation from './documentation/idsintehittapa-profanity-dictionary-1.0.0-resolved.json'
+
+// Defines the port the app will run on.
+import express from 'express'
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// homepage
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(documentation)
 })
 
-// Start the server
+
+app.get('/curses', (req, res) => {
+  const { language, category_id } = req.query
+  let curseList = profanityDictionary
+  
+  // query by category_id
+  if (category_id) {
+    curseList = curseList.filter((curses) => curses.category_id === +category_id)
+  }
+
+  // query by language
+  if (language) {
+    curseList = curseList.filter((curses) => curses.language.toString().toLocaleLowerCase().includes(language.toLocaleLowerCase()))
+  }
+
+  // display total nrs of items in array
+  const totalCurses = curseList.length
+
+  // adding pagination using slice
+  const page = req.query.page ?? 0 //otherwise 0 as default 
+  const pageSize = req.query.pageSize ?? 20 //otherwise 20 as default 
+
+  // calculate start index
+  const startIndex = page * pageSize
+
+  // calculate and bound the end index
+  const endIndex = startIndex + pageSize
+
+  const cursesPerPage = curseList.slice(startIndex, endIndex)
+  const returnObject = { totalNumCurses: totalCurses, startIndex: startIndex, endIndex: endIndex, numCurses: cursesPerPage.length, cursesPerPage}
```

Comment:
Since some of the keys and values are the same:
`const returnObject = { totalNumCurses: totalCurses, startIndex, endIndex, numCurses: cursesPerPage.length, cursesPerPage}`